components/playlist/browser/playlist_service.cc

```diff
@@ -322,6 +322,25 @@ void PlaylistService::NotifyPlaylistChanged(mojom::PlaylistEvent playlist_event,
     obs.OnPlaylistStatusChanged({playlist_event, playlist_id});
 }
 
+void PlaylistService::NotifyMediaFilesUpdated(
+    const GURL& url,
+    std::vector<mojom::PlaylistItemPtr> items) {
+  if (items.empty()) {
```

Comment:
Should we still fire the listeners if there are no items (i.e. the site removed all the videos).

Reply:
For now, I guess we don't need to, as this event is only about dynamically added files. On the other hands, if the listeners want to make it sure, they can explicitly call another method, then they'll get an empty list.